packages/open-normal-extension/open_normal.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Chromium browser extension action script to open a URL in normal browser
+# Uses stdin and stdout for communicating with browser extension API
+# First four bytes tell the size of payload, and rest is the payload,
+# which is standard json data. Same format is used for input and output.
+
+# Configuration file which contains the givc parameters
+CFGF="/etc/open-normal-extension.cfg"
+
+# Get integer (ASCII code) from a byte
+function Ord {
+    printf "%d" "\"$1"
+}
+
+# Get byte from an integer (ASCII code)
+function Chr {
+    printf "%b" "$(printf "\\\\x%02x" "$1")"
+}
+
+# Send browser extension API message. First send four bytes of the length of
+# the payload and then the actual payload.
+function Msg {
+    local len b1 b2 b3 b4
+
+    len="${#1}"
+    b1="$(( len & 255 ))"
+    b2="$(( (len >> 8) & 255 ))"
+    b3="$(( (len >> 16) & 255 ))"
+    b4="$(( (len >> 24) & 255 ))"
+    Chr "$b1"
+    Chr "$b2"
+    Chr "$b3"
+    Chr "$b4"
+    printf "%s" "$1"
+}
+
+# Read the four bytes of length in the received API message
+LANG=C IFS= read -r -d '' -n 1 B1
+LANG=C IFS= read -r -d '' -n 1 B2
+LANG=C IFS= read -r -d '' -n 1 B3
+LANG=C IFS= read -r -d '' -n 1 B4
+
+# If read byte is zero, the corresponding variables are unset
+# Convert bytes to their ASCII values or zero if unset
+if [ -z "$B1" ]; then
+    B1=0
+else
+    B1="$(Ord "$B1")"
+fi
+if [ -z "$B2" ]; then
+    B2=0
+else
+    B2="$(Ord "$B2")"
+fi
+if [ -z "$B3" ]; then
+    B3=0
+else
+    B3="$(Ord "$B3")"
+fi
+if [ -z "$B4" ]; then
+    B4=0
+else
+    B4="$(Ord "$B4")"
+fi
+
+# Convert the four bytes to a integer length
+# Using multiply instead of bit shifing as bit shifts broke syntax highlighting
+LEN="$((B1+(B2*256)+(B3*65536)+(B4*16777216)))"
```

Comment:
Could also use `od`: `read LEN _ < <(od -t u4 -A n -N 4 --endian=little)`

Reply:
Agreed, that would also work, and it would be short and tidy. It would have a dependency though, but Gnu coreutils package is probably already there. Above considerations apply here too.

Reply:
I decided that using od is justified as it makes the script a lot simpler.
I did however ditch the read in this case: `LEN="$(od -t u4 -A n -N 4 --endian=little)"`